crates/extensions/tedge_config_manager/src/actor.rs

```diff
@@ -103,73 +105,55 @@ impl ConfigManagerActor {
             plugin_config,
             pending_operations: HashMap::new(),
             input_receiver,
-            mqtt_publisher,
+            output_sender,
             download_sender,
             upload_sender,
         }
     }
 
-    async fn process_mqtt_message(&mut self, message: MqttMessage) -> Result<(), ChannelError> {
-        match ConfigOperation::request_from_message(&self.config, &message) {
-            Ok(Some(ConfigOperation::Snapshot(request))) => match request.status {
-                CommandStatus::Init => {
+    async fn process_operation_request(
+        &mut self,
+        request: ConfigOperation,
+    ) -> Result<(), ChannelError> {
+        match request {
+            ConfigOperation::Snapshot(topic, request) => match request.status {
+                CommandStatus::Init | CommandStatus::Scheduled => {
```

Comment:
A command has to be executed either on `init` or the `scheduled` state.

- Init is the trigger when the actor is directly exposed to MQTT (in the unit tests notably).
- When the actor is running behind the workflow actor, the trigger is the scheduled state.